### ISSUE TITLE
Expression parser fails to parse constant matrix literals

### ISSUE THREAD

**Issue by jongleur1983** [ClosedXML/ClosedXML#588]
**Do you want to request a *feature* or report a *bug*?**
- [ ] Bug
- [x] Feature

**Version of ClosedXML**
current develop

**What is the current behavior?**

The following formula is valid in an excel cell: 
=MDETERM({3,6,1;1,1,0;3,10,2})

The parameter of the MDETERM function here is a constant encoding for the following matrix:
3, 6, 1
1, 1, 0
3, 10, 2 

ClosedXML fails to parse the expression and throws an ExpressionParseException when reading the opening curly brace {.

**What is the expected behavior or new feature?**
the sample input provided above should be parsed successfully and it should be possible to get the result as a matrix.

When that's achieved, functions like MDETERM could get completed to accept constant matrices as well.

**Did this work in previous versions of our tool?  Which versions?**

No

## Reproducibility
**This is an important section. Read it carefully. Failure to do so will cause a 'RTFM' comment.**

Without a code sample, it is unlikely that your issue will get attention. Don't be lazy. Do the effort and assist the developers to reproduce your problem. Code samples should be [minimal complete and verifiable](https://stackoverflow.com/help/mcve). Sample spreadsheets should be attached whenever applicable. Remove sensitive information.

**Code to reproduce problem:**
```c#
XLWorkbook.EvaluateExpr("MDETERM({3,6,1;1,1,0;3,10,2})");
```

**Comment by igitur** [comment]
Not sure if you want to look at it, but I won't soon. 

Planning to replace the formula parser completely with XLParser, which can parser that.

**Comment by jongleur1983** [comment]
Thanks for the info.
You're referring to https://github.com/spreadsheetlab/XLParser ?

**Comment by igitur** [comment]
Yup

**Comment by jongleur1983** [comment]
ok. Working on integrating it already? As the parsing issues with the old parser are a blocker for completing unit tests for some functions that would be useful - and if not, it might be something I could look into myself.

**Comment by igitur** [comment]
No, haven't really started with it. I've made some contributions to XLParser itself to prepare it for .NET Standard compatibility. That's all.

**Comment by jongleur1983** [comment]
I see your PR there to support the dotnet core stuff directly.
As that's not gone through I'll try to get it integrated using the old nuget source in my branch to integrate the parser itself already.
I hope the switch once your PR is merged will be easy (using the default nuget source then)

**Comment by jongleur1983** [comment]
@igitur biggest problem currently for introducing XLParser is that ClosedXML uses strong naming while XLParser does not. Is far as I understand this means that XLParser cannot be just referenced, but has to be included in the ClosedXML project.
Is there a good reason to use strong naming at all? 
It's quite outdated and doesn't help a lot.

**Comment by vbjay** [comment]
Would you want to know that closedxml is exactly the way it was written?  I
would. It would suck to have a malicious entity put a malicious copy of
closedxml in the program.

**Comment by jongleur1983** [comment]
@vbjay well, to trust that I would have to check the signature with the package maintainer over a side channel that's secure and independent from the usual channel, and if I that check passes I "only" have to trust that person that the original package is not malicious.
So there are two steps involved that are dangerous, even with signed packages.

Do you trust any nuget package just because it's on nuget.org, is signed and the code is on github?
Did you EVER check the public key of a strong named package you used? Do you require that key to be signed? Signed by whom?

That's like PGP Mails with signature, but without checking the signature - just not useful.

As I neither double check full source code, nor back-check developers of packages or do the effort to securely check the correctness of a signing key even for signed libraries:
I don't even care ClosedXML is exactly the way it was written. I care about it being good.
Of course I don't want it to be broken or malicious, but signing doesn't much here.

**Comment by vbjay** [comment]
The verification of signing is done on runtime. If your application was
compiled against assembly a and I put b in it's place, the strong name
handles that problem.

**Comment by vbjay** [comment]
Since we write Excel files. It is a needed solution.

**Comment by jongleur1983** [comment]
ah, ok - that's an argument ;)
Nevertheless: I have no idea how to reference XLParser than as that package is not signed (yet?).
If someone could help me here I'd like to look into it again later.
Thanks for the explanations so far.

**Comment by vbjay** [comment]
You can download the source of xl parser and strong name it. You could
submit a pull request. You could ask them to do what a lot of publishers
do, publish 2 packages one being strong named.

**Comment by igitur** [comment]
I was planning to change ClosedXML to non-signed with the `netstandard2.0` merge. 
This is the post that persuaded me. I trust that blogger. https://www.pedrolamas.com/2016/03/01/still-strong-naming-your-assemblies-you-do-know-its-2016-right/

**Comment by jongleur1983** [comment]
;) thanks, that solves my concerns, then I'll try to proceed and remove the strong naming in my branch.
